src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/UsePropertyInsteadOfCountMethodWhenAvailableTests.cs

```diff
@@ -741,6 +741,29 @@ End Class
 ",
             }.RunAsync();
         }
+
+        [Fact]
+        [WorkItem(4652, "https://github.com/dotnet/roslyn-analyzers/issues/4652")]
+        public static async Task FixCausesCircularity_NoDiagnostic()
+        {
+            var code = @"
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+class X : IEnumerable<int>
+{
+    public int Count => this.Count(); // Don't suggest using Count property because it'll be endless recursion.
+
+    public IEnumerator<int> GetEnumerator() => throw new NotImplementedException();
+
+    IEnumerator IEnumerable.GetEnumerator() => throw new NotImplementedException();
+}
```

Comment:
I wasn't able to write a parallel VB code. [Here](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBsQDdhJiaMA+AkmAA4D2CMUABAMoCescYAsAFDHmU0NNgB0AYTIoUcAMYwAlmQB2UdpwpU6jeAOGiJ0uVH4BxOLMRTxi0sp5rm/ADJTZAR3btBKaDQAa7ar+qcxMCMVQgBRWQBXIIQIYDEACgB5ADN/WXgAc0QASh8/PN8SCLjTagAFBDISRBh6amEI9OoAQRpCdLgshGoAXmoAWTghMkaYeNy2AvLilFKAMUbJGVlqQxhwqMQIGApxlraN6O3dlLTMnP9SQOCDyKO4uCTU9vOEbIM4dbutnYQpv2oABUABaVADu1AAcnAIZCyDAAsxgnAACahAAe4jgJB0snGU3CKOoC1kSzkLjYAIqUnQ2zgxMWuP8hy2DwA+msWTFfntWszvtyKJcSNd0rdNjEHh8vhLjn9KQC/CDwVCYVD4Yigh00Zjsbj8QrfISGaTcexjW4PEA==) is my attempt to write it.

However, the call to `Me.Count()` in my code is referring to the property, not the LINQ extension.